ivy-prescient.el

```diff
@@ -111,19 +111,31 @@ This is for use in `ivy-re-builders-alist'."
 (defvar ivy-prescient--old-re-builder nil
   "Previous default value in `ivy-re-builders-alist'.")
 
+(defun ivy-prescient--elements-ensure (element)
+  "Ensure that the type of `ELEMENT' is acceptable to prescient.
+In addition to string, ivy accepts many types of data, so some
+processing is needed to ensure that `prescient' can handle them."
+  (if (stringp element)
+      element
+    (cond ((symbolp element) (symbol-name element))
+          ((consp element) (symbol-name (car element)))
+          ((listp element) (car element)))))
```

Comment:
I think the third case will never trigger, because the only thing that satisfies `listp` but not `consp` is `nil`. So I think this would break the existing case where you have a cons cell whose car is a string, because it would try to take `symbol-name` of a string?

Reply:
Adjusting the order will solve this problem,right? I'll put the `listp` to the first case.